### ISSUE TITLE
Lint/NestedMethodDefinition shouldn't flag singleton methods

### ISSUE THREAD

**Issue by CyborgMaster** [bbatsov/rubocop#2704]
I have a function that returns an object, and decorates it with a patched method.  This doesn't fall under the category of this cop because it doesn't create a method in the outer scope.  

Flagged code:
```ruby
  def medium_mock
    mock('medium') do |m|
      stubs(:get_search_ranges).returns([Object.new])
      stubs(:deliver)
      def m.select_fields(param)
        param
      end
      stubs type: 'test'
    end
  end
```
Offense:
```
Lint/NestedMethodDefinition: Method definitions must not be nested. Use lambda instead.
      def m.select_fields(param)
      ^^^^^^^^^^^^^^^^^^^^^^^^^^
```

```
$ rubocop -V
0.36.0 (using Parser 2.3.0.1, running on ruby 2.2.2 x86_64-darwin14)
```
